src/content/url-handling.js

```diff
@@ -101,7 +101,13 @@ function toDataSourceEnum(str: string): DataSource {
   throw new Error('unexpected data source');
 }
 
-export function stateFromLocation(location: Location): URLState {
+type MockLocation = {|
+  pathname: string,
+  search: string,
+  hash: string,
+|};
```

Comment:
If you remove the strictness here, you could type `location` as MockLocation only and avoid the OR. Flow would be happy to accept `Location` for `{ pathname: string, search: string, hash: string }` thanks to width subtyping.